src/Util.ts

```diff
@@ -1,49 +1,29 @@
+// tslint:disable-next-line:no-reference
+/// <reference path="global.d.ts" />
+
 import BigNumber from 'bignumber.js';
-import fetch from 'node-fetch';
 import * as Web3 from 'web3';
 import * as Web3WsProvider from 'web3-providers-ws';
 
 import { Networks } from './Enum';
-import { IBlock, IServices, ITxRequest } from './Types';
-
-function apiCall(url: string): Promise<any> {
-  return fetch(url).then((response: any) => {
-    if (!response.ok) {
-      return new Error(response.statusText);
-    }
-    return response.json();
-  });
-}
-
-const SERVICES: IServices = {
-  //should return GWEI value of the gasPrice
-  [Networks.Mainnet]: [
-    {
-      api: 'https://dev.blockscale.net/api/gasexpress.json',
-      field: 'standard',
-      morph: (val: any): number => {
-        const GWEI = 10e9;
-        return Number(val * GWEI);
-      }
-    },
-    {
-      api: 'https://ethgasstation.info/json/ethgasAPI.json',
-      field: 'average',
-      morph: (val: any): number => {
-        const GWEI = 10e9;
-        return Number(val * GWEI) / 10;
-      }
-    }
-  ]
+import {
+  BlockScaleFetchingService,
+  EthGasStationFetchingService,
+  EthGasStationInfo
+} from './GasEstimation';
+import { IBlock, ITxRequest, GasPriceEstimation } from './Types';
+
+const GAS_PRICE_FETCHING_SERVICES = {
+  [Networks.Mainnet]: {
+    blockScale: new BlockScaleFetchingService(),
+    ethGasStation: new EthGasStationFetchingService()
```

Comment:
I think we should add a defaultProvider here that uses eth.getGasPrice. By doing so we can get rid of treating this as an extra backup provider.

Additionally I would represent that in Array instead of object, in Object the order of keys is no guaranteed, I guess alphanumerical or something, here we want to prioritize the provider blockScale or ethGasStation.

Reply:
I like the suggestion with `defaultProvider`!

Reply:
I like the suggestion with `defaultProvider`!

Reply:
The non-determinism of using the Object seemed fine to me since the two estimates should be pretty close most of the time and it would be interesting if different TimeNodes are using slightly different estimates. I guess the best solution is to let the TimeNode choose which they prefer. Overall it doesn't really matter because if one fails then the other will activate.

Also not sure why to make a `defaultProvider` for the `web3` call. We treat it like a back-up because it is a back-up. Also makes it easy for the external fetching call to just fail on networks other than the mainnet, since we don't have data for those networks and estimates matter less. It seems like adding `defaultProvider` would be just wrapping over the `web3` call and maybe I don't see why that's more clear.